### PR TITLE
feat: add Uniswap v3

### PR DESCRIPTION
**Pull request by nledo**

### CODE REVIEW

File: sdk/src/protocols/uniswap_v3/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        allow.mainnet.uniswap_v3.positions_nft.refundETH(),
+        allow.mainnet.uniswap_v3.positions_nft.unwrapWETH9(undefined, c.avatar),
+        allow.mainnet.uniswap_v3.positions_nft.collect(
+          {
+            tokenId: nftIds ? oneOf(nftIds) : undefined,
+            recipient: ZERO_ADDRESS
+          }
+        ),
+        allow.mainnet.uniswap_v3.positions_nft.sweepToken(
+          undefined,
```
[the remark this answers is not in the document]
Comment (jfschwarz) [reply sdk/src/protocols/uniswap_v3/index.ts:93]: I don't understand `sweepToken` well enough to give a good answer...

**Review by jfschwarz — COMMENTED** [review]
I realized we're now forcing users to specify the `avatar` parameter while its sole purpose is for validating the given NFT IDs. I believe it would be better to get rid of the param and just trust the user-provided NFT IDs.

There could even be scenarios where you wanna enable permissions for NFTs not currently owner by the avatar, e.g. increasing liquidity of positions held by in a different account or setting up permissions to manage NFTs that are yet to be transferred to the avatar address.

**Comment by jfschwarz** [comment]
@nledo run `yarn fix:prettier` to make the CI workflow green.

**Comment by nledo** [comment]
@jfschwarz finally we decided to leave the `avatar` parameter since we consider that the recipient for the mint, collect, unwrapETH9 and sweepToken should be the avatar. I removed the `validateNftIds` to allow the possible scenarios you stated here:
### There could even be scenarios where you wanna enable permissions for NFTs not currently owner by the avatar, e.g. increasing liquidity of positions held in a different account or setting up permissions to manage NFTs that are yet to be transferred to the avatar address.

File: sdk/src/protocols/uniswap_v3/index.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      ...allowErc20Approve(tokensForTargets || [], [
+        contracts.mainnet.uniswap_v3.positions_nft,
+      ]),
+      ...allowErc20Approve(mintTokenAddresses, [
+        contracts.mainnet.uniswap_v3.positions_nft,
+      ]),
+      {
+        ...allow.mainnet.uniswap_v3.positions_nft.mint(
+          {
+            recipient: avatar,
```
Comment (jfschwarz) [inline sdk/src/protocols/uniswap_v3/index.ts:59]:
```suggestion
            recipient: c.avatar,
```

This makes it fully dynamic (avatar address is looked up when checking permissions onchain) and allows us to entirely get rid of this function's `avatar` param.

Comment (nledo) [reply sdk/src/protocols/uniswap_v3/index.ts]: You're totally right!

**Pull request merged**